### PR TITLE
Allow disabling reader's zoom out

### LINKED ISSUE #62: Allow disabling reader zoom out
### Describe your suggested feature

I've never intentionally zoomed out whilst reading. It would be great to have the option to disable reader zoom out.

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have updated the app to version **[0.16.0](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Splintorien**
Closes #62

### CODE REVIEW

**Comment by AntsyLich** [comment]
Make it a button to disable zoom out

**Comment by AntsyLich** [comment]
Current implementation: Enable (on by default)
I want: Disable (off by default)

**Comment by AntsyLich** [comment]
Umm

**Comment by Splintorien** [comment]
> Umm

I wanted updated my fork but it did some weird things, all good

**Comment by AntsyLich** [comment]
Next time make cha

**Comment by AntsyLich** [comment]
Oops

File: app/src/main/java/eu/kanade/presentation/reader/settings/GeneralSettingsPage.kt [hunk 1]
```diff
@@ -73,4 +73,9 @@ internal fun ColumnScope.GeneralPage(screenModel: ReaderSettingsScreenModel) {
         label = stringResource(MR.strings.pref_flash_page),
         pref = screenModel.preferences.flashOnPageChange(),
     )
+
+    CheckboxItem(
+        label = stringResource(MR.strings.pref_disable_zoom_out),
+        pref = screenModel.preferences.disableZoomOut(),
+    )
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/presentation/reader/settings/GeneralSettingsPage.kt:80]: Should be in [ReadingModePage.kt](https://github.com/mihonapp/mihon/blob/6bdb37be65757ca903c5c2a249ca03331b04d673/app/src/main/java/eu/kanade/presentation/reader/settings/ReadingModePage.kt) under `WebtoonViewerSettings`

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/setting/ReaderPreferences.kt [hunk 2]
```diff
@@ -72,6 +72,8 @@ class ReaderPreferences(
 
     fun skipDupe() = preferenceStore.getBoolean("skip_dupe", false)
 
+    fun disableZoomOut() = preferenceStore.getBoolean("disable_zoom_out", false)
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/setting/ReaderPreferences.kt:75]:
```suggestion
    fun webtoonDisableZoomOut() = preferenceStore.getBoolean("webtoon_disable_zoom_out", false)
```

File: i18n/src/commonMain/resources/MR/base/strings.xml [hunk 3]
```diff
@@ -453,6 +453,7 @@
     <string name="pref_high">High</string>
     <string name="pref_low">Low</string>
     <string name="pref_lowest">Lowest</string>
+    <string name="pref_disable_zoom_out">Disable zoom out</string>
```
Comment (AntsyLich) [inline i18n/src/commonMain/resources/MR/base/strings.xml:456]:
```suggestion
    <string name="pref_webtoon_disable_zoom_out">Disable zoom out</string>
```

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/webtoon/WebtoonRecyclerView.kt [hunk 4]
```diff
@@ -33,6 +33,15 @@ class WebtoonRecyclerView @JvmOverloads constructor(
     private var firstVisibleItemPosition = 0
     private var lastVisibleItemPosition = 0
     private var currentScale = DEFAULT_RATE
+    var zoomOutDisabled = false
+        set (value) {
+            field = value
+            if (!value) {
+                zoom(currentScale, DEFAULT_RATE, x, 0f, y, 0f)
+            }
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/webtoon/WebtoonRecyclerView.kt:41]: Only invoke this if `currentScale` is less than `DEFAUT_RATE`

**Review by AntsyLich — CHANGES_REQUESTED** [review]

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/webtoon/WebtoonRecyclerView.kt [hunk 5]
```diff
@@ -33,6 +33,15 @@ class WebtoonRecyclerView @JvmOverloads constructor(
     private var firstVisibleItemPosition = 0
     private var lastVisibleItemPosition = 0
     private var currentScale = DEFAULT_RATE
+    var zoomOutDisabled = false
+        set (value) {
+            field = value
+            if (currentScale < DEFAULT_RATE) {
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/webtoon/WebtoonRecyclerView.kt:39]:
```suggestion
            if (value && currentScale < DEFAULT_RATE) {
```

File: app/src/main/java/eu/kanade/presentation/reader/settings/ReadingModePage.kt [hunk 6]
```diff
@@ -197,6 +197,11 @@ private fun ColumnScope.WebtoonViewerSettings(screenModel: ReaderSettingsScreenM
         label = stringResource(MR.strings.pref_double_tap_zoom),
         pref = screenModel.preferences.webtoonDoubleTapZoomEnabled(),
     )
+
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/presentation/reader/settings/ReadingModePage.kt:200]:
```suggestion
```

File: app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsReaderScreen.kt [hunk 7]
```diff
@@ -341,8 +341,11 @@ object SettingsReaderScreen : SearchableSettings {
                 Preference.PreferenceItem.SwitchPreference(
                     pref = readerPreferences.webtoonDoubleTapZoomEnabled(),
                     title = stringResource(MR.strings.pref_double_tap_zoom),
-                    enabled = true,
                 ),
+                Preference.PreferenceItem.SwitchPreference(
+                    pref = readerPreferences.webtoonDisableZoomOut(),
+                    title = stringResource(MR.strings.pref_webtoon_disable_zoom_out),
+                )
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsReaderScreen.kt:348]:
```suggestion
                ),
```

**Review by AntsyLich — CHANGES_REQUESTED** [review]

**Pull request merged**